src/main/scala/apps/convolution.scala

```diff
@@ -14,13 +14,37 @@ import shine.OpenCL.KernelExecutor._
 object convolution {
   private val id = fun(x => x)
 
+  private val dotElemWeightsHL = fun((weights, elem) =>
+    zip(join(elem))(weights) |>
+    map(separableConvolution2D.mulT) |>
+    reduce(add)(l(0.0f))
+  )
+
   private val dotElemWeights = fun((weights, elem) =>
     oclReduceSeqUnroll(AddressSpace.Private)(fun((acc, pair) => {
       val pixel = pair._1
       val weight = pair._2
       acc + (pixel * weight)
     }))(l(0.0f))(zip(join(elem))(weights)))
 
+  // FIXME: could not find original Lift expression, this is made up
+  val blurXHL: ToBeTyped[Expr] = depFun((n: Nat) => fun(
+    (n `.` n `.` f32) ->: (17 `.` f32) ->: (n `.` n `.` f32)
+  )((matrix, weights) =>
+    map(map(dotElemWeightsHL(weights)))
+    o slide2D(1, 1, 17, 1)
+    o padClamp2D(0, 0, 8, 8) $ matrix
+  ))
+
+  // FIXME: could not find original Lift expression, this is made up
+  val blurYHL: ToBeTyped[Expr] = depFun((n: Nat) => fun(
+    (n `.` n `.` f32) ->: (17 `.` f32) ->: (n `.` n `.` f32)
+  )((matrix, weights) =>
+    map(map(transpose >> dotElemWeightsHL(weights)))
+    o slide2D(17, 1, 1, 1)
+    o padClamp2D(8, 8, 0, 0) $ matrix
```

Comment:
I know we don't have any guidance on this, but the change in this file is introducing two styles:
 - the piping `x |> f |> g` style, vs.
 - the traditional functional `f o g $ x` style.
 
  Should we aim to commit to a preferred style?
 
 I personally much prefer the piping style because reads, left-to-right, top-to-bottom which I believe is much simpler.

Reply:
I also prefer the piping style, the question is if we want to keep the style that was in original Lift or not (it makes expression comparison easier when looking for potential differences / mistakes).

Reply:
I see. I would say: we should prefer the piping style by default and we can use the other one if we have a good reason for that. I would also say, that it would make sense to add a brief comment on why that style is used so that we don't refactor it by accident away.

Reply:
I changed one or two expression to the piping style. Otherwise I am fine if the other expressions get refactored later on, this was just easier short-term. Maybe we can merge as-is and people are free to refactor to the piping style if and when they desire?